### ISSUE TITLE
Error while scaffolding project: failed to initialize project with "go.kubebuilder.io/v3"

### ISSUE THREAD

**Issue by sagar-jadhav** [operator-framework/operator-sdk#4355]
#### What did you do?

- Installed `operator-sdk` via `Homebrew`. Followed this [steps](https://sdk.operatorframework.io/docs/installation/#install-from-homebrew-macos)
- Scaffolded project using the following command:

```
operator-sdk init --domain=test.com --repo=<REPO>
```

#### What did you expect to see?

The project should be initialized successfully without any error.

#### What did you see instead? Under which circumstances?

```
FATA[0000] failed to initialize project with "go.kubebuilder.io/v3": only the go.mod and files with the prefix "(.)" are allowed before the init 
``` 

#### Environment

**Operator type:**

/language go

**Kubernetes cluster type:**

OpenShift

`$ operator-sdk version`

v1.3.0

`$ go version` (if language is Go)

go1.15.5

`$ kubectl version`

v1.19.4

**Comment by zhangtbj** [comment]
I also see this problem by using the latest `v1.3.0` version. And didn't find a solution for that.

**Comment by sagar-jadhav** [comment]
With the below command, it is working now:

```
operator-sdk init --domain=test.com --repo=<REPO> --plugins go.kubebuilder.io/v2
```

looks like some issue with the `go.kubebuilder.io/v3` plugin.

**Comment by zhangtbj** [comment]
Using `--plugins go.kubebuilder.io/v2` also works for me.

Thanks @sagar-jadhav !

**Comment by camilamacedo86** [comment]
Hi @sagar-jadhav and @zhangtbj, 

The problem here is that you are using the tool to scaffold a project in a directory that has files. See the error message:  

> failed to initialize project with "go.kubebuilder.io/v3": **only the go.mod and files with the prefix "(.)" are allowed before the init** 

This validation was added for v3/ projects and a truly recommend you use v3 instead of v2 for any new project and do NOT scaffold new projects in a directory that has files already. It can bring issues and problems which were faced in the past and because of this it is no longer allowed. 

I am closing this one as solved. However, please feel free to ping us and/or create new issues if you need.

**Comment by Phiph** [comment]
Hi Camila, 

I'm trying to run the init on a folder that just has a .git folder in side of it. I think it may be a false positive. As I think my folder structure meets the validation requirements. 

![image](https://user-images.githubusercontent.com/8458416/103795841-63f30f00-503e-11eb-8615-b3308b93ee21.png)

**Comment by alecmerdler** [comment]
I see this as well when I have a `.git` directory. Removing the `.git` directory and re-running `operator-sdk init` fixed the issue.

**Comment by object88** [comment]
Reproduced with `operator-sdk` v1.4.  Created a folder, ran `git init`, then `go mod init`, and `operator-sdk init` failed.

**Comment by lmcdasm** [comment]
Hello All.

As as side question, does this mean that we have to build our things now outside of git, and then put them into the folder so we can do version control of the operators?  seems a real pain to have such a hook.

**Comment by SYS48** [comment]
same issue here, I ran ini command in a folder with only `.git/` and `.gitignore`

**Comment by camilamacedo86** [comment]
Hi @SYS48, 

It is solved for mater and will be solved for the next release.

**Comment by nk15786k** [comment]
Hello All.
For me it is resolved by setting the GO111MODULE to "on".

**Comment by deepy** [comment]
I'm getting this error in a directory that only has a `.idea` folder (from goland)

**Comment by estroz** [comment]
As @camilamacedo86 said, this is resolved in the next release (v1.5.0).

**Comment by rgolangh** [comment]
Make sure the directory is completely clean. In my case `rm -rf .git` and it completed with no problems.
The needs to be more informative though.
